### DISCUSSION TITLE
Integrate with OpenWebUI?

### DISCUSSION THREAD

**Discussion by mdlmarkham** [assafelovic/gpt-researcher#821] [Ideas]
This is an awesome project! What do you think about integrating with [OpenWebUI](https://github.com/open-webui/open-webui) - it would be awesome if you could use the output of -researcher as input to a model chat or file library.

**Comment by ElishaKay** [comment]
Welcome @mdlmarkham 

I've played with OpenWebUI a bit & I think I see where you're coming from regarding the complementary nature of the 2 projects.

I definitely like the idea of integrating those LLM's in their catalogue into GPTR.

Didnt understand the use cases you mentioned but we're open to ideas.

**Reply by mdlmarkham** [reply]
Let me flesh out the use case a little more... I typically use OWUI for conversations with agents about topics, e.g. how to configure something in my homelab, or how to design an information system that I'm building. I've integrated OWUI with [SearXNG](https://github.com/searxng/searxng) and I can have it do some basic research... but it's not very robust. What I would like to be able to do is to formulate a research question during my conversation in OWUI - pass that research question to a dedicated team of researchers in GPTR - have them execute the research task and then pass the results back to the original agent / conversation that is in-progress in OWUI. Ideally there would be a review step, and then follow-up research tasks may be generated, or the final product could be stored in a document library for reference / context.

**Reply by ElishaKay** [reply]
Sounds like you'll want to dig in deeper into the Open Web UI docs.

https://docs.openwebui.com/tutorial-development/

Try to find something about webhooks, or injecting custom JavaScript into an Open Web UI Flow.

In order to send a webhook message to GPTR, have a look here at [gptr-webhook.js](https://github.com/assafelovic/gpt-researcher/blob/b5457d4e98721f5e38aa903d682d4189fa62f013/docs/discord-bot/gptr-webhook.js)

And here's how to leverage that file:
```
const { sendWebhookMessage } = require('./gptr-webhook');
// Await the response from GPTR via WebSocket
let gptrResponse = await sendWebhookMessage('what is the meaning of life?');
```

May the force be with you
